qa/rpc-tests/p2p-autoinstantsend.py

```diff
@@ -196,88 +90,37 @@ def set_autoix_spork_state(self, state):
-                input = {}
-                input["txid"] = tx['txid']
-                input['vout'] = tx['vout']
-                in_amount += float(tx['amount'])
-                inputs.append(input)
-            elif in_amount > amount:
-                break
-            elif len(inputs) < max_inputs:
-                input = {}
-                input["txid"] = tx['txid']
-                input['vout'] = tx['vout']
-                in_amount += float(tx['amount'])
-                inputs.append(input)
-            else:
-                input = {}
-                input["txid"] = tx['txid']
-                input['vout'] = tx['vout']
-                in_amount -= last_amount
-                in_amount += float(tx['amount'])
-                inputs[-1] = input
-            last_amount = float(tx['amount'])
-
-        assert(len(inputs) > 0)
-        assert(in_amount > amount)
-        # fill outputs
-        receiver_address = node_to.getnewaddress()
-        change_address = node_from.getnewaddress()
-        fee = 0.001
-        outputs={}
-        outputs[receiver_address] = amount
-        outputs[change_address] = in_amount - amount - fee
-        rawtx = node_from.createrawtransaction(inputs, outputs)
-        return node_from.signrawtransaction(rawtx)
-
-    def check_IX_lock(self, txid):
-        # wait for instantsend locks
-        start = time()
-        locked = False
-        while True:
-            is_trx = self.nodes[0].gettransaction(txid)
-            if is_trx['instantlock']:
-                locked = True
-                break
-            if time() > start + 10:
-                break
-            sleep(0.1)
-        return locked
-
     # sends regular IX with high fee and may inputs (not-simple transaction)
     def send_regular_IX(self):
         receiver_addr = self.nodes[self.receiver_idx].getnewaddress()
         txid = self.nodes[0].instantsendtoaddress(receiver_addr, 1.0)
-        return self.check_IX_lock(txid)
+        return self.check_IX_lock(txid, self.nodes[0])
 
 
     # sends simple trx, it should become IX if autolocks are allowed
     def send_simple_tx(self):
         raw_tx = self.create_raw_trx(self.nodes[0], self.nodes[self.receiver_idx], 1.0, 1, 4)
         txid = self.nodes[0].sendrawtransaction(raw_tx['hex'])
         self.sync_all()
-        return self.check_IX_lock(txid)
+        return self.check_IX_lock(txid, self.nodes[0])
 
     # sends complex trx, it should never become IX
     def send_complex_tx(self):
         raw_tx = self.create_raw_trx(self.nodes[0], self.nodes[self.receiver_idx], 1.0, 5, 100)
         txid = self.nodes[0].sendrawtransaction(raw_tx['hex'])
         self.sync_all()
-        return self.check_IX_lock(txid)
+        return self.check_IX_lock(txid, self.nodes[0])
 
     def run_test(self):
+        # feed the sender with some balance
+        sender_addr = self.nodes[self.sender_idx].getnewaddress()
+        self.nodes[0].sendtoaddress(sender_addr, 100)
```

Comment:
Why amount changed from 1 to 100?

Reply:
The test wasn't stable with old values, I'll check this issue one more time

Reply:
Like I said 

> I'd like to keep refactoring and fixing issues (if that was the reason) in separate PRs

;)

Reply:
The order of the operations during network setup was changed,  that's why `create_raw_transaction` fails much more often than initially after this changes. I can fix this error in separate PR and after its merging I can rebase this PR over it,  so it can be only refactoring without changing any functionality. Is it ok?

Reply:
I've added https://github.com/dashpay/dash/pull/2353 that fixes this error,  after its merging I'll change this PR.